File: app/src/main/java/com/sysu/edu/widget/RecentClassWidgetWorker.java
package com.sysu.edu.widget;

import android.appwidget.AppWidgetManager;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class RecentClassWidgetWorker extends Worker {
    
    public RecentClassWidgetWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }
    
    @NonNull
    @Override
    public Result doWork() {
        getApplicationContext().startService(new Intent(getApplicationContext(), RecentClassWidget.class).setAction(AppWidgetManager.ACTION_APPWIDGET_UPDATE)
                .putExtra(AppWidgetManager.EXTRA_APPWIDGET_IDS, AppWidgetManager.getInstance(getApplicationContext())
                        .getAppWidgetIds(new ComponentName(getApplicationContext(), RecentClassWidget.class))));
        return Result.success();
    }
}

File: app/src/main/java/com/sysu/edu/widget/RecentClassWidgetWorker.kt
package com.sysu.edu.widget

import android.appwidget.AppWidgetManager
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import androidx.work.Worker
import androidx.work.WorkerParameters

class RecentClassWidgetWorker(context: Context, workerParams: WorkerParameters) :
	Worker(context, workerParams) {
	override fun doWork(): Result {
		applicationContext.startService(Intent(applicationContext, RecentClassWidget::class.java).setAction(AppWidgetManager.ACTION_APPWIDGET_UPDATE)
												 .putExtra(AppWidgetManager.EXTRA_APPWIDGET_IDS, AppWidgetManager.getInstance(applicationContext)
													 .getAppWidgetIds(ComponentName(applicationContext, RecentClassWidget::class.java))))
		return Result.success()
	}
}
